src/main/java/com/exadel/project/internship/controller/PostedInternshipController.java

```diff
@@ -0,0 +1,45 @@
+package com.exadel.project.internship.controller;
+
+
+import com.exadel.project.common.exception.EntityNotFoundException;
+import com.exadel.project.internship.dto.InternshipDTO;
+import com.exadel.project.internship.dto.InternshipDetailsDTO;
+import com.exadel.project.internship.service.InternshipService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("internship")
+@RequiredArgsConstructor
+public class PostedInternshipController {
+
+    private final InternshipService internshipService;
+    private static final String ID = "/{id}";
+
+    @GetMapping
+    public ResponseEntity<List<InternshipDTO>> findPostedInternships(@RequestParam(value = "search",
+            required = false) String search, @RequestParam(value = "sort", required = false) String sort){
+        return ResponseEntity.ok(internshipService.getAllPosted(search, sort));
+    }
+
+    @GetMapping(value = ID)
+    public ResponseEntity<InternshipDetailsDTO> findPostedInternshipById(@PathVariable ("id") Long id) throws EntityNotFoundException {
+        return ResponseEntity.ok(internshipService.getPostedById(id));
+    }
+
+    @GetMapping(value = ID + "/disapprove")
+    public ResponseEntity<InternshipDetailsDTO> findAndPostUnpostedInternshipById(@PathVariable ("id") Long id) throws EntityNotFoundException {
+        return ResponseEntity.ok(internshipService.disapproveUnpostedById(id));
+    }
```

Comment:
Why we need such endpoint? Maybe it's better to have one generic get by specification endpoint?

Reply:
fixed by one PATCH endpoint with one param, which should be equal to "VISIBLE_FOR_ADMINS" or "VISIBLE_FOR_INTERNS"